### PR TITLE
[INTEL oneDNN] OneDNN Refactoring Part 1: Remove obsolete operations

### PR DESCRIPTION
**Pull request by gzmkl**
This PR removes obsolete oneDNN operations (oneDNN blocked format related) as well as MKL layout pass tests and conversion tests (test cases are all related to blocked format).

Results:
   1. 12 source files are deleted
   2. 8 source files are update (bazel files, tests)
   3. Total 8843 lines of code are cleaned away!

### CODE REVIEW

**Review by sagunb — APPROVED** [review]

**Review by penpornk — COMMENTED** [review]
Thank you for the PR and sorry for the late review! I'm unsure about removing op registrations right away and wanted to check.

@sagunb Thank you for the quick review! To [remove normal ops](https://www.tensorflow.org/guide/versions#remove_or_restrict_an_ops_functionality), we'll have to mark them for deprecation and wait for the next major release before removing the actual functionality. I'm not sure about ops with `_` name prefix. 

Is it okay to just remove the registrations and implementations for these ops? These `_Mkl` ops aren't saved in any SavedModels anyway. They are only introduced through graph rewrites in C++ layer.

**Comment by sagunb** [comment]
Adding @rohan100jain as a second pair of eyes for this. I think this should be an ok removal since these ops should not be end user facing (only introduced in the internal graph rewrite passes) and so should not affect any end user behaviour but perhaps there are some other restrictions I am overlooking.

**Comment by rohan100jain** [comment]
Do we expect these ops to ever show up in a SavedModel? If not, I think its fine to remove them.

**Comment by rohan100jain** [comment]
To be safe one option would be to hollow out the kernel implementation and just throw an error. That could also lead to a fair amount of code deletion?

**Comment by penpornk** [comment]
Thank you for the quick reply, @rohan100jain!

> Do we expect these ops to ever show up in a SavedModel? If not, I think its fine to remove them.

No, these ops can't show up in a SavedModel (their names all start with the `_` prefix). They are only introduced in the C++ layer through graph rewrites.

> To be safe one option would be to hollow out the kernel implementation and just throw an error. That could also lead to a fair amount of code deletion?
> We leave it like that for one release and next release if no one complains, we can then do the full scale deletion

Sounds good to me. :)

@gzmkl Would you mind changing the PR to keep all op registrations for now and adding a TODO to remove each of them in TF 2.13 instead? All `REGISTER_KERNEL_BUILDER` of these ops can register the same kernel that will just throw an error (op is deprecated). Thank you very much!

**Comment by gzmkl** [comment]
I will follow up with your suggestion soon

**Comment by gzmkl** [comment]
@penpornk @rohan100jain Per PR review suggestion, the ops registration of deprecated oneDNN ops has been added. 
This delta change will be removed in future TensorFlow release once we get Google's signal.  THANKS

**Comment by gzmkl** [comment]
Suggestions have been taken. Please check latest commit

**Review by penpornk — COMMENTED** [review]
Thank you for the changes!

It seems we haven't prevented `mkl_layout_pass` from inserting these deprecated nodes yet if people build with `--config=mkl`. Should we remove `--define=enable_mkl=true` from all configs?

File: tensorflow/core/kernels/mkl/mkl_deprecated_ops.cc
Comment (gzmkl) [reply tensorflow/core/kernels/mkl/mkl_deprecated_ops.cc]: Code change done per suggestions.

**Comment by gzmkl** [comment]
@penpornk  I do not think that we need to remove --define=enable_mkl=true from all configs. 
I consulted this with Ramesh and rest of my team. TF with oneDNN enabled now only supports native format
(ref> NativeFormatEnabled() already returns true, graph/mkl_graph_util.cc).  oneDNN ops with label kMklLayoutDependentOpLabel are no longer used for graph node rewrite (in TF 2.11 or even earlier version. 

In my part 2 refactoring PR (pending on this one), all related push_back calls of deprecated ops will be removed.
For this PR, even though these ops are push_back'ed to rinfo_ list (mkl_layout_pass.cc), they are not used for node rewrite (no matter if there is --config=mkl build option). 

Thanks!

**Review by penpornk — APPROVED** [review]
Thank you for the changes and for the detailed explanation!

**Review by sagunb — APPROVED** [review]

**Comment by penpornk** [comment]
Manually closing this since it has been merged in https://github.com/tensorflow/tensorflow/commit/f623c4f155b08011484c8a9ec568e7194cc690a6.

**Comment by gzmkl** [comment]
@penpornk  Hi Penporn, I wonder why this PR was closed. Its branch has not been merged to master. 
Thanks!

**Comment by penpornk** [comment]
@gzmkl Hi Guozhong, I already had the fix ready soon after the revert, but I have to wait until the branch cut is done before I can get it in since this is a non-fix PR and we are in the freeze period. I'll get this PR back after the branch cut. You don't need to do anything. :)

I'll also add a point in the TF 2.12 release notes that we are deprecating these internal ops in TF 2.13.

**Comment by gzmkl** [comment]
@penpornk THANK you for your help!   -GZ

**Comment by gzmkl** [comment]
@penpornk  Hopefully we can merge this PR to the master branch ASAP, as our team is starting to working oneDNN 3.0 integration. Due to oneDNN 3.0 is not API backward compatible, temporally oneDNN 2.7.x and 3.0 API will used in TensorFlow integration (similar strategy as we did from oneDNN 0.x to 1.x 3 years ago).  Thanks!

**Comment by penpornk** [comment]
@gzmkl Sorry for the delay! The PR has been brought back in https://github.com/tensorflow/tensorflow/commit/155b34bad741124f778978f806d82551f9d35a66

**Pull request closed without merging**
